CollectNewspaperKarel.java

```diff
@@ -1,17 +1,45 @@
-/*
- * File: CollectNewspaperKarel.java
- * --------------------------------
- * At present, the CollectNewspaperKarel subclass does nothing.
- * Your job in the assignment is to add the necessary code to
- * instruct Karel to walk to the door of its house, pick up the
- * newspaper (represented by a beeper, of course), and then return
- * to its initial position in the upper left corner of the house.
- */
-
-import stanford.karel.*;
-
-public class CollectNewspaperKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: CollectNewspaperKarel.java
+ * --------------------------------
+ * At present, the CollectNewspaperKarel subclass does nothing.
+ * Your job in the assignment is to add the necessary code to
+ * instruct Karel to walk to the door of its house, pick up the
+ * newspaper (represented by a beeper, of course), and then return
+ * to its initial position in the upper left corner of the house.
+ */
+
+import stanford.karel.*;
+
+public class CollectNewspaperKarel extends SuperKarel {
+
+	public void run() {
+		turnRight();
+		move();
+		turnLeft();
+		fastRun();
+		pickMyBeeper();
+		returnHome();
+
+	}
+	
+	private void fastRun() {
+		move();
+		move();
+		move();
+	}
+	
+	private void pickMyBeeper() {
+	while (beepersPresent()) {

```

Comment:
You don't need to have a while loop, from the assignment you are guarantee that there will be one beeper at this position. All you need to do is: pickBeeper();